File: zig/isogram/isogram.zig
const std = @import("std");
pub fn isIsogram(str: []const u8) bool {
    var set = std.bit_set.IntegerBitSet(26).initEmpty();
    var count_not_alpha: u16 = 0;

    for (str) |c| {
        if (!std.ascii.isAlphabetic(c)) {
            count_not_alpha += if (c == ' ' or c == '-') 1 else 0;
        } else {
            const lower = std.ascii.toLower(c);
            set.set(lower - 'a');
        }
    }

    return set.count() == str.len - count_not_alpha;
}

File: python/isogram/isogram.py
def is_isogram(string: str) -> bool:
    list_str: list[str] = []

    for char in string.lower():
        if char == " " or char == "-":
            continue

        if len(list_str) == 0:
            list_str.append(char)
        else:
            if char in list_str:
                return False
            else:
                list_str.append(char)

    return True
